src/credentials/attestation_verifier/TestDACRevocationDelegateImpl.cpp

```diff
@@ -0,0 +1,205 @@
+        {
+            continue;
+        }
+        for (const auto & revokedSerialNumber : revokedSet["revoked_serial_numbers"])
+        {
+            if (revokedSerialNumber.asString() == serialNumber)
+            {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+CHIP_ERROR TestDACRevocationDelegateImpl::GetAKIDHexStr(const ByteSpan & certDer, MutableCharSpan & outAKIDHexStr)
+{
+    uint8_t akidBuf[kAuthorityKeyIdentifierLength];
+    MutableByteSpan akid(akidBuf);
+
+    ReturnErrorOnFailure(ExtractAKIDFromX509Cert(certDer, akid));
+
+    return BytesToHexStr(akid, outAKIDHexStr);
+}
+
+CHIP_ERROR TestDACRevocationDelegateImpl::GetSerialNumberHexStr(const ByteSpan & certDer, MutableCharSpan & outSerialNumberHexStr)
+{
+    uint8_t serialNumberBuf[kMaxCertificateSerialNumberLength] = { 0 };
+    MutableByteSpan serialNumber(serialNumberBuf);
+
+    ReturnErrorOnFailure(ExtractSerialNumberFromX509Cert(certDer, serialNumber));
+    return BytesToHexStr(serialNumber, outSerialNumberHexStr);
+}
+
+CHIP_ERROR TestDACRevocationDelegateImpl::GetIssuerNameBase64Str(const ByteSpan & certDer,
+                                                                 MutableCharSpan & outIssuerNameBase64String)
+{
+    uint8_t issuerBuf[kMaxCertificateDistinguishedNameLength] = { 0 };
+    MutableByteSpan issuer(issuerBuf);
+
+    ReturnErrorOnFailure(ExtractIssuerFromX509Cert(certDer, issuer));
+    VerifyOrReturnError(outIssuerNameBase64String.size() >= BASE64_ENCODED_LEN(issuer.size()), CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    uint32_t encodedLen = Base64Encode32(issuer.data(), static_cast<uint32_t>(issuer.size()), outIssuerNameBase64String.data());
+    outIssuerNameBase64String.reduce_size(encodedLen);
+    return CHIP_NO_ERROR;
+}
+
+bool TestDACRevocationDelegateImpl::IsCertificateRevoked(const ByteSpan & certDer)
+{
+    static constexpr uint32_t maxIssuerBase64Len = BASE64_ENCODED_LEN(kMaxCertificateDistinguishedNameLength);
+
+    char issuerNameBuffer[maxIssuerBase64Len]                            = { 0 };
+    char serialNumberHexStrBuffer[2 * kMaxCertificateSerialNumberLength] = { 0 };
+    char akidHexStrBuffer[2 * kAuthorityKeyIdentifierLength]             = { 0 };
+
+    MutableCharSpan issuerName(issuerNameBuffer);
+    MutableCharSpan serialNumber(serialNumberHexStrBuffer);
+    MutableCharSpan akid(akidHexStrBuffer);
+
+    VerifyOrReturnValue(CHIP_NO_ERROR == GetIssuerNameBase64Str(certDer, issuerName), false);
+    ChipLogDetail(NotSpecified, "Issuer: %.*s", static_cast<int>(issuerName.size()), issuerName.data());
+
+    VerifyOrReturnValue(CHIP_NO_ERROR == GetSerialNumberHexStr(certDer, serialNumber), false);
+    ChipLogDetail(NotSpecified, "Serial Number: %.*s", static_cast<int>(serialNumber.size()), serialNumber.data());
+
+    VerifyOrReturnValue(CHIP_NO_ERROR == GetAKIDHexStr(certDer, akid), false);
+    ChipLogDetail(NotSpecified, "AKID: %.*s", static_cast<int>(akid.size()), akid.data());
```

Comment:
Add a TODO (and create associated issue) to cross-validate the CRLSignerCertificate and CRLSignerDelegator per spec:

```
. If a matching `RevocationSet` is found, determined if `serialNumber` is a member of the set.
    .. If the entity type whose revocation status is being verified is a PAI, then the subject PAI's issuer (a PAA) SHALL fulfill one of the following two cases, otherwise return immediately:
      ... The Subject and Subject Key of the PAI certificate's issuer matches exactly the CRLSignerCertificate's subject (i.e. the CRLSignerCertificate is a PAA).
      ... The Subject and Subject Key of the PAI certificate's issuer matches exactly the PAA which is the issuer of the CRLSignerCertificate (i.e. the CRLSignerCertificate is a CRL signer delegated by a PAA).
    .. If the entity type whose revocation status is being verified is a DAC, then:
      ... If the `CRLSignerDelegator` is present, then the subject DAC's issuer (a PAI) SHALL match the `CRLSignerDelegator` in both Subject Key and Subject, otherwise return immediately.
      ... If the `CRLSignerDelegator` is absent, then the subject DAC's issuer (a PAI) SHALL match the `CRLSignerCertificate` in both Subject Key and Subject, otherwise return immediately.
```

Those checks are not currently do, and MUST be done later once this is supported in revocation set construction.

Reply:
Added a TODO and an issue: #34587